### PR TITLE
Add 'number of nodes' restriction to license

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: blackbox/docs/admin/system-information.rst [hunk 1]
```diff
@@ -1421,7 +1421,7 @@ Here's an example query::
   |  1 | The setting 'discovery.zen.minimum_master_nodes' must not be ... |
   |  2 | The total number of partitions of one or more partitioned tab... |
   |  3 | The following tables need to be upgraded for compatibility wi... |
-  |  6 | Your CrateDB license is not close to expiry. Enjoy CrateDB!      |
+  |  6 | Your CrateDB license is fine. Enjoy CrateDB!                     |
```
Comment (smakalias) [inline blackbox/docs/admin/system-information.rst]: could we change 'fine' to 'valid' or may be rephrase it to something like "You have a valid CrateDB license. Enjoy CrateDB"?

File: sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java [hunk 2]
```diff
@@ -27,34 +27,41 @@
...
 import static io.crate.expression.reference.sys.check.SysCheck.Severity.MEDIUM;
 import static io.crate.license.LicenseExpiryNotification.MODERATE;
 
 @Singleton
-public class LicenseExpiryCheck implements SysCheck {
+public class LicenseCheck implements SysCheck {
 
     private static final int ID = 6;
-    private static final String LICENSE_NOT_CLOSE_TO_EXPIRY_DESCRIPTION = "Your CrateDB license is not close to expiry. Enjoy CrateDB!";
+    private static final String LICENSE_OK = "Your CrateDB license is fine. Enjoy CrateDB!";
```
Comment (smakalias) [inline sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java]: same as above

File: sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java [hunk 3]
```diff
@@ -63,24 +70,39 @@ public boolean validate() {
...
+                return false;
+
+            case MAX_NODES_VIOLATED:
+                severity = HIGH;
+                description = getLinkedDescription(ID, String.format(
+                    Locale.ENGLISH,
+                    "The license is limited to %d nodes, but there are %d nodes in the cluster. " +
+                    "To upgrade your license visit https://crate.io/license-update/",
+                    licenseService.currentLicense().maxNumberOfNodes(),
+                    clusterService.state().getNodes().getSize()
```
Comment (smakalias) [inline sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java]: this is not consistent with the implementation which checks for data nodes!

File: sql/src/test/java/io/crate/expression/reference/sys/check/cluster/LicenseCheckTest.java [hunk 4]
```diff
@@ -40,16 +40,16 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-public class LicenseExpiryCheckTest extends CrateUnitTest {
+public class LicenseCheckTest extends CrateDummyClusterServiceUnitTest {
 
     private LicenseService licenseService;
-    private LicenseExpiryCheck expirationCheck;
+    private LicenseCheck expirationCheck;
```
Comment (smakalias) [inline sql/src/test/java/io/crate/expression/reference/sys/check/cluster/LicenseCheckTest.java]: could we also change the field name?

File: sql/src/test/java/io/crate/expression/reference/sys/check/cluster/LicenseCheckTest.java [hunk 5]
```diff
@@ -96,7 +99,7 @@ public void testLessThanOneDayToExpiryTriggersSevereCheck() {
     @Test
     public void testCheckIsAlwaysValidWhenEnterpriseIsDisabled() {
         Settings settings = Settings.builder().put("license.enterprise", false).build();
-        LicenseExpiryCheck expiryCheckNoEnterprise = new LicenseExpiryCheck(settings, mock(LicenseService.class));
+        LicenseCheck expiryCheckNoEnterprise = new LicenseCheck(settings, mock(LicenseService.class), clusterService);
         assertThat(expiryCheckNoEnterprise.validate(), is(true));
     }
```
Comment (smakalias) [inline sql/src/test/java/io/crate/expression/reference/sys/check/cluster/LicenseCheckTest.java]: should we also add a test for the max_node case?

File: sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java [hunk 6]
```diff
@@ -27,34 +27,41 @@
...
 
     @Inject
-    public LicenseExpiryCheck(Settings settings, LicenseService licenseService) {
+    public LicenseCheck(Settings settings, LicenseService licenseService, ClusterService clusterService) {
         enterpriseEnabled = SharedSettings.ENTERPRISE_LICENSE_SETTING.setting().get(settings);
-        description = (enterpriseEnabled) ? LICENSE_NOT_CLOSE_TO_EXPIRY_DESCRIPTION : LICENSE_NA_COMMUNITY_DESCRIPTION;
+        this.clusterService = clusterService;
+        if (!enterpriseEnabled) {
+            description = LICENSE_NA_COMMUNITY_DESCRIPTION;
+        }
```
Comment (seut) [inline sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java]: as `description()` is not marked as `Nullable`, couldn't this change result in a NPE on the CE?
